sycl/include/syclcompat/device.hpp

```diff
@@ -331,22 +473,12 @@ class device_ext : public sycl::device {
     // Guard the destruct of current_queues to make sure the ref count is safe.
     lock.lock();
   }
-  queue_ptr create_queue(bool print_on_async_exceptions = false,
-                         bool in_order = true) {
+  sycl::queue *create_queue(bool print_on_async_exceptions = false,
+                            bool in_order = true) {
     std::lock_guard<std::mutex> lock(m_mutex);
-    sycl::property_list prop = {};
-    if (in_order) {
-      prop = {sycl::property::queue::in_order()};
-    }
-    if (print_on_async_exceptions) {
-      _queues.push_back(std::make_shared<sycl::queue>(
-          _ctx, *this, detail::exception_handler, prop));
-    } else {
-      _queues.push_back(std::make_shared<sycl::queue>(_ctx, *this, prop));
-    }
-    return _queues.back().get();
+    return create_queue_impl(print_on_async_exceptions, in_order);
   }
-  void destroy_queue(queue_ptr &queue) {
+  void destroy_queue(sycl::queue *&queue) {
```

Comment:
Whats the motivation for this change from queue_ptr -> sycl:queue *?
I think it is a bit less readable, even if it is the same underneath.